### ISSUE TITLE
Update current theme header when a new theme is selected

### ISSUE THREAD

**Issue by JorgeMucientes** [woocommerce/woocommerce-android#10398]
We currently don't refresh the "Current theme" section after selecting a new theme from the Theme Picker.

https://github.com/woocommerce/woocommerce-android/assets/2663464/fe8cbaa7-2e46-4dcc-8208-b9a2dccb05a2

**Comment by hichamboushaba** [comment]
Good catch @JorgeMucientes, should we instead just close the theme picker in this case or do you think it's better to keep the user on the screen and just refresh the current theme?

**Comment by JorgeMucientes** [comment]
Hey @hichamboushaba, good point. 
To me, it feels more natural to return to the theme picker where I'm coming from and where I can confirm the changes were applied. The navigation feels more consistent that way. However, I don't have a strong opinion. Is there any reason you might prefer redirecting the user to settings?

**Comment by hichamboushaba** [comment]
@JorgeMucientes I don't have a strong opinion on it, but these are the reasons I suggested it:
1. The user changed the theme and saved it, so they should be done with the theme picker.
2. It's simpler to implement and requires way fewer changes than adding logic to reload current theme.

> I'm coming from and where I can confirm the changes were applied

We show a Snackbar, so they user will still be able to confirm the changes were applied.

> The navigation feels more consistent that way

Can you elaborate on this? the `popUpTo` will behave same as navigation UP 🤔

**Comment by JorgeMucientes** [comment]
> Can you elaborate on this? The `popUpTo` will behave the same as navigation UP 🤔 

@hichamboushaba When clicking on a list item from the list I'm opening a detail view. So when navigating back from it, whether it is because I tap back or because I selected/installed the theme, I expect to land on the list screen I came from. That's usually how list -> detail view navigation works. There are exceptions to this of course, that's why I don't have a strong opinion. 

> It's simpler to implement and requires way fewer changes than adding logic to reload current theme.

Yeah is true. Although the code is simple enough anyway, and it's already implemented now 😅. So, in terms of effort, the work is done. I think we can move forward with refreshing the header after installing the theme.
